### PR TITLE
[ML] Model Monitoring Out of Box Experience + Spark + Final PuP Contract Changes

### PR DESCRIPTION
**Pull request by needuv**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_monitoring/definition.py [hunk 1]
```diff
@@ -83,18 +89,41 @@ def _to_rest_object(self) -> RestMonitorDefinition:
...
+
+    def _populate_default_signal_information(
+        self,
+        model_inputs_arm_id: str,
+        model_inputs_type: str,
+        model_outputs_arm_id: str,
+        model_outputs_type: str,
+    ):
+        self.monitoring_signals = {
+            DefaultMonitorSignalNames.DATA_DRIFT_SIGNAL: DataDriftSignal._get_default_data_drift_signal(
```
Comment (mastloui-msft) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_monitoring/definition.py:120]: just to make sure, this creates the default collection but only gets used if the fields aren't set right?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_monitoring/signals.py [hunk 2]
```diff
@@ -375,6 +419,32 @@ def _from_rest_object(cls, obj: RestMonitoringDataQualitySignal) -> "DataDriftSi
...
+        return cls(
+            target_dataset=TargetDataset(
+                dataset=MonitorInputData(
+                    input_dataset=Input(
+                        path=production_data_id,
+                        type=production_data_type,
+                    ),
+                    dataset_context=MonitorDatasetContext.MODEL_INPUTS,
+                ),
+                lookback_period=60,
```
Comment (mastloui-msft) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_monitoring/signals.py:435]: I believe the default is 7 days - Let's double check with bo

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_monitoring/definition.py (hunk 1 above)
Comment (needuv) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_monitoring/definition.py:120]: yes, this only gets created if a deployment id is given and there are no signals

**Review by luigiw — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/monitoring/monitor_definition.py [hunk 3]
```diff
@@ -16,12 +16,13 @@
...
 from azure.ai.ml._schema.monitoring.alert_notification import AlertNotificationSchema
-from azure.ai.ml._schema.core.fields import NestedField, UnionField, ComputeField, StringTransformedEnum
+from azure.ai.ml._schema.core.fields import NestedField, UnionField, StringTransformedEnum
 from azure.ai.ml._schema.core.schema import PatchedSchemaMeta
+from azure.ai.ml._schema.spark_resource_configuration import SparkResourceConfigurationSchema
 
 
 class MonitorDefinitionSchema(metaclass=PatchedSchemaMeta):
-    compute = ComputeField()
+    compute = NestedField(SparkResourceConfigurationSchema)
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/monitoring/monitor_definition.py:25]: It's unfortunate we have to make a `compute` filed complex, which breaks the consistency with jobs.

Comment (needuv) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/monitoring/monitor_definition.py:25]: Agreed, but unfortunately amlcompute isn't supported. We should probably just rename this to something like `spark_configuration` for the public preview refresh.

**Pull request merged**